sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -37,6 +37,32 @@ export async function sendRequest(
 ): Promise<HttpResponse> {
   const httpClient = customHttpClient ?? getCachedDefaultHttpsClient();
   const request = buildPipelineRequest(method, url, options);
+  if (options) {
+    if (options.timeout) {
+      request.timeout = options.timeout;
+    }
+
+    if (options.onUploadProgress) {
+      request.onUploadProgress = options.onUploadProgress;
+    }
+
+    if (options.onDownloadProgress) {
+      request.onDownloadProgress = options.onDownloadProgress;
+    }
+
+    if (options.allowInsecureConnection) {
+      request.allowInsecureConnection = true;
+    }
```

Comment:
This maybe could be removed, as we have handled allowInsecureConnection when buildPipelineRequest.